backend/src/main.py

```diff
@@ -0,0 +1,19 @@
+import os
+
+import sqlalchemy
+from dotenv import load_dotenv
+from fastapi import FastAPI
+
+from .handlers import location
+
+load_dotenv()
+
+app = FastAPI()
+
+app.include_router(location.router)
+
+
+@app.on_event("startup")
+def startup_event():
+    global conn
+    conn = sqlalchemy.create_engine(os.getenv("DATABASE_URL") or "")
```

Comment:
this just provides a way for the backend to interact with the database through a connection pool, any input sanitization can still be done between the handlers and the db

Reply:
OK, I think I meant something more along the lines of do we want any references to a specific DB library (sqlalchemy) inside the same file where we handle API calls, as that would make it more difficult to switch to another DB library if that was necessary in the future (as in, we would need to replace all code specific to sqlalchemy). I think I just watched too much Uncle Bob Clean Code recently.

Regardless, I think this is something we can talk about later (and is not super important in a project of this scale) and I'll approve this so I don't block the merge.

Reply:
it might be nice to uncouple a library from our code, but if we end up doing that we will need to write a ton of code for compatibility, and ultimately our code base is small enough that we can just rewrite any orm calls, which would probably take less work than rewriting any compatibility code

Reply:
I agree, looks good!